docs/arcaflow/workflows/expressions.md

```diff
@@ -260,6 +260,7 @@ Functions:
 | `toLower(string)`                                | string       | Returns the input with Unicode letters mapped to their lower case.                                                                                                                                                                                                                                                                                                                                                                                                                             |
 | `toUpper(string)`                                | string       | Returns the input with Unicode letters mapped to their upper case.                                                                                                                                                                                                                                                                                                                                                                                                                             |
 | `splitString(string, string)`                    | list[string] | Returns a list of the substrings which appear between instances of the specified separator; the separator instances are not included in the resulting list elements; adjacent occurrences of separator instances as well as instances appearing at the beginning or ending of the input will produce empty string list elements.<br>&nbsp; Param 1: The string to split.<br>&nbsp; Param 2: The separator.                                                                                     |
+| `readFile(string)`                    | string | Returns a string representation of a file.<br>&nbsp; Param 1: The filepath string of the file to read into memory.<br>&nbsp;                                                                                     |
 
```

Comment:
I don't really like the word "representation".  I recommend "contents".  Also, since there is only one parameter, it would be nicer if we could avoid numbering it and having to treat it as a list with one thing in it.  Perhaps something like, "Given string containing a file path, returns the contents of the indicated file as a string."

Reply:
`Return the contents of a file as a string`?

Obvious questions that might be nice to answer here might include: in what context is a relative file path resolved; can you use environment variables like `$HOME/file`?

Reply:
Yes you can use `$HOME`. The golang code calls [filepath.Clean()](https://cs.opensource.google/go/go/+/go1.22.1:src/path/filepath/path.go;l=99) which delegates to OS filesystem specific strategies.

Reply:
I would have expected that `$HOME` would only work if there were a shell (or similar) involved in the file access...is that really the case, here?

FWIW, `filepath.Clean("${HOME}/mydir/myfile.txt")` returns `${HOME}/mydir/myfile.txt` which doesn't look all that "clean" to me, and `filepath.Clean("${HOME}/../myneighbor/mydir/myfile.txt")` returns `myneighbor/mydir/myfile.txt` which looks outright wrong....  😒  (And, `filepath.Abs()` doesn't make out much better.)

So, I think Dave's question boils down to, what is the current directory to which relative paths would be relative (and would the workflow writer know that?)?

Reply:
Indeed, according to the documentation for `path.Clean`:

> Clean returns the shortest path name equivalent to path by purely lexical processing. It applies the following rules iteratively until no further processing can be done:
>
> 1. Replace multiple [Separator](https://pkg.go.dev/path/filepath#Separator) elements with a single one.
> 1. Eliminate each . path name element (the current directory).
> 1. Eliminate each inner .. path name element (the parent directory) along with the non-.. element that precedes it.
> 1. Eliminate .. elements that begin a rooted path: that is, replace "/.." by "/" at the beginning of a path.

It's not clear that any of the `filepath` methods know anything about environment variables, which is probably useful to mention; and `Clean`, in particular, doesn't even resolve relative file paths. (`Abs` does. I assume that [`os.FileRead`](https://pkg.go.dev/os#ReadFile) does, and possibly handles environment variables as well, though the documentation says nothing about either...)

Reply:
You are correct, the golang code does not expand shell environment variables.

Reply:
I will mention that relative filepaths must be written to be relative to the working directory from which the arcaflow process is executed.